### PR TITLE
feat: Azure DevOps PR E2E pipeline

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What this PR does / why we need it**: Add E2E with Azure DevOps pipeline to eventually replace circle CI.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**If applicable**:
- [ ] documentation
- [ ] unit tests
- [ ] tested backward compatibility (ie. deploy with previous version, upgrade with this branch)

**Release note**:

```release-note
```

### CODE REVIEW

File: .pipelines/e2e-job-template.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    modulePath: '$(GOPATH)/src/github.com/$(build.repository.name)' # Path to the module's code
+    TIMEOUT: 20m
+    CLUSTER_DEFINITION: ${{ parameters.apimodel }}
+    ORCHESTRATOR_RELEASE: ${{ parameters.k8sRelease }}
+    CREATE_VNET: ${{ parameters.createVNET }}
+    CLEANUP_ON_EXIT: true
+    CLEANUP_IF_FAIL: true
+    RETAIN_SSH: false
+    ENABLE_KMS_ENCRYPTION: true
+    SUBSCRIPTION_ID: '$(SUBSCRIPTION_ID_E2E_KUBERNETES)'
```
Comment (tariq1890) [inline .pipelines/e2e-job-template.yaml]: Lets rename this to AZURE_SUBSCRIPTION_ID. This seems to be the env naming convention

File: .pipelines/e2e-step-template.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      shopt -s extglob
+      mv !(gopath) '$(modulePath)'
+      echo '##vso[task.prependpath]$(GOBIN)'
+      echo '##vso[task.prependpath]$(GOROOT)/bin'
+    displayName: 'Set up the Go workspace'
+  - script: make build-binary
+    displayName: Compile
+    workingDirectory: $(modulePath)
+  - script: |
+      export CLIENT_ID=$(SERVICE_PRINCIPAL_CLIENT_ID_E2E_KUBERNETES)
```
Comment (tariq1890) [inline .pipelines/e2e-step-template.yaml]: AZURE_CLIENT_ID?

File: .pipelines/e2e-job-template.yaml (hunk 1 above)
Comment (CecileRobertMichon) [reply .pipelines/e2e-job-template.yaml]: This is defined in by our e2e tests https://github.com/Azure/aks-engine/blob/master/test/e2e/azure/cli.go#L35 and is the same as circle uses

File: .pipelines/e2e-step-template.yaml (hunk 2 above)
Comment (CecileRobertMichon) [reply .pipelines/e2e-step-template.yaml]: Same here: this is defined in by our e2e tests https://github.com/Azure/aks-engine/blob/master/test/e2e/azure/cli.go#L35 and is the same as circle uses

File: .pipelines/e2e-job-template.yaml (hunk 1 above)
Comment (tariq1890) [reply .pipelines/e2e-job-template.yaml]: Then let's do away this `SUBSCRIPTION_ID_E2E_KUBERNETES`. I assume that variable is being declared in vsts internally. Let's do that internal declaration for `SUBSCRIPTION_ID` itself.

Comment (CecileRobertMichon) [reply .pipelines/e2e-job-template.yaml]: Sure. I just copied that over from circle. I will still need to declare the var here for it to be available in the container job. I assume the idea behind e2e_kubernetes was that we have different variables subscriptions declared in circle env vars. For now this is the only one in vsts though so I'm okay with changing it.

Comment (tariq1890) [reply .pipelines/e2e-job-template.yaml]: Actually makes sense. We can have this PR as is for now

**Comment by CecileRobertMichon** [comment]
installing sudo is failing as described in https://github.com/dotnet/dotnet-docker/issues/788... I think we need sudo in the image, or get rid of install kubectl altogether and put that in the image.

File: .pipelines/pr-e2e.yaml [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    workingDirectory: $(modulePath)
+
+- template: e2e-job-template.yaml
+  parameters:
+    name: 'default_linux_e2e'
+    apimodel: 'examples/kubernetes.json'
+
+- template: e2e-job-template.yaml
+  parameters:
+    name: 'k8s_1dot10_release_e2e'
```
Comment (tariq1890) [inline .pipelines/pr-e2e.yaml]: nit: can we have an underscore instead of `dot`? More readable that way IMO

Comment (jackfrancis) [reply .pipelines/pr-e2e.yaml]: If we don't like "dot" I vote for a period (`.`)

Comment (CecileRobertMichon) [reply .pipelines/pr-e2e.yaml]: period isn't a valid char in the name (that was my first choice too)

I can change it to underscore

Comment (tariq1890) [inline .pipelines/pr-e2e.yaml]:
```suggestion
     name: 'k8s_1_10_release_e2e'
```

Comment (CecileRobertMichon) [reply .pipelines/pr-e2e.yaml]: why the extra space?

Comment (tariq1890) [reply .pipelines/pr-e2e.yaml]: Sorry. Edited the suggestion

**Pull request merged**